### PR TITLE
Fixes #436 and improve Bgm code a bit

### LINKED ISSUE #436: audioPlayer null exception
void didChangeAppLifecycleState(AppLifecycleState state) {
    if (state == AppLifecycleState.resumed) {
      if (isPlaying &&
          audioPlayer != null &&
          audioPlayer.state == AudioPlayerState.PAUSED) {
        audioPlayer.resume();
      }
    } else {
      **audioPlayer.pause();**
    }
  }

null

### PR DESCRIPTION
**Pull request by luanpotter**
# Description

Fixes #436 and improve Bgm code a bit

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `develop`
- [x] This PR is targeted to merge into `develop` (not `master`)
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: lib/bgm.dart [hunk 1]
```diff
@@ -43,51 +43,48 @@ class Bgm extends WidgetsBindingObserver {
   ///
   /// It is safe to call this function even when a current BGM track is
   /// playing.
-  void play(String filename, {double volume}) async {
+  Future<void> play(String filename, {double volume}) async {
```
Comment (luanpotter) [inline lib/bgm.dart:46]: just making all methods consistent (async) and allowing callers to await if they want. this should not be breaking.

File: lib/bgm.dart [hunk 2]
```diff
@@ -110,13 +107,11 @@ class Bgm extends WidgetsBindingObserver {
...
     if (state == AppLifecycleState.resumed) {
-      if (isPlaying &&
-          audioPlayer != null &&
-          audioPlayer.state == AudioPlayerState.PAUSED) {
+      if (isPlaying && audioPlayer?.state == AudioPlayerState.PAUSED) {
         audioPlayer.resume();
       }
     } else {
-      audioPlayer.pause();
+      audioPlayer?.pause();
```
Comment (luanpotter) [inline lib/bgm.dart:114]: this is the bug fix

**Review by spydon — COMMENTED** [review]
Lgtm!

**Review by renancaraujo — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,7 @@
 
 ## [next]
  - Adding MouseMovementDetector
+ - Fixes #436 and improve Bgm code a bit
```
Comment (erickzanardo) [inline CHANGELOG.md:5]: I would go with a smaller and more formal sentence here like just: `Fixes BGM error`

**Review by erickzanardo — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (luanpotter) [reply CHANGELOG.md:5]: I wouldn't say that is "more formal" haha but I will replace

**Pull request merged**